newm-server/src/main/kotlin/io/newm/server/aws/MdcSetup.kt.kt

```diff
@@ -0,0 +1,32 @@
+package io.newm.server.aws
+
+import org.slf4j.MDC
+import aws.sdk.kotlin.services.ec2.Ec2Client
+import aws.sdk.kotlin.services.ec2.model.DescribeInstancesRequest
+import org.koin.core.qualifier.named
+
+suspend fun setInstanceId() {
+    val instanceId = getInstanceIdFromAws() // Replace with your method to get the instance ID
+    MDC.put("instanceId", instanceId)
+}
+
+suspend fun getInstanceIdFromAws(): String? {
+    var instanceId: String? = null
+    val awsRegion = named("aws.region")
+
+    val request =
+        DescribeInstancesRequest {
+            maxResults = 6
+        }
+
+    Ec2Client { region = awsRegion.toString() }.use { ec2 ->
```

Comment:
Don't create an `Ec2Client` here. We already have an instance in the dependency injection graph. Use that one via Koin `inject()`